source/extensions/transport_sockets/tls/cert_validator/default_validator.cc

```diff
@@ -340,46 +340,41 @@ ValidationResults DefaultCertValidator::doVerifyCertChain(
                                 SSL_CTX_get0_param(&ssl_ctx))) {
       OPENSSL_PUT_ERROR(SSL, ERR_R_X509_LIB);
       const char* error = "verify cert failed: init and setup X509_STORE_CTX";
-      onVerifyError(ssl_extended_info, error);
-      return {ValidationResults::ValidationStatus::Failed, absl::nullopt, error};
+      stats_.fail_verify_error_.inc();
+      ENVOY_LOG(debug, error);
+      return {ValidationResults::ValidationStatus::Failed,
+              Envoy::Ssl::ClientValidationStatus::Failed, absl::nullopt, error};
     }
     const bool verify_succeeded = (X509_verify_cert(ctx.get()) == 1);
 
     if (!verify_succeeded) {
       const std::string error =
           absl::StrCat("verify cert failed: ", Utility::getX509VerificationErrorInfo(ctx.get()));
-      onVerifyError(ssl_extended_info, error);
+      stats_.fail_verify_error_.inc();
+      ENVOY_LOG(debug, error);
       if (allow_untrusted_certificate_) {
-        return ValidationResults{ValidationResults::ValidationStatus::Successful, absl::nullopt,
+        return ValidationResults{ValidationResults::ValidationStatus::Successful,
+                                 Envoy::Ssl::ClientValidationStatus::Failed, absl::nullopt,
                                  absl::nullopt};
       }
       return {ValidationResults::ValidationStatus::Failed,
+              Envoy::Ssl::ClientValidationStatus::Failed,
               SSL_alert_from_verify_result(X509_STORE_CTX_get_error(ctx.get())), error};
     }
-    if (ssl_extended_info) {
-      ssl_extended_info->setCertificateValidationStatus(
-          Envoy::Ssl::ClientValidationStatus::Validated);
-    }
+    detailed_status = Envoy::Ssl::ClientValidationStatus::Validated;
   }
   std::string error_details;
   uint8_t tls_alert = SSL_AD_CERTIFICATE_UNKNOWN;
-  const bool succeeded = verifyCertAndUpdateStatus(
-      ssl_extended_info, leaf_cert, transport_socket_options.get(), &error_details, &tls_alert);
+  const bool succeeded = verifyCertAndUpdateStatus(leaf_cert, transport_socket_options.get(),
+                                                   detailed_status, &error_details, &tls_alert);
   return succeeded ? ValidationResults{ValidationResults::ValidationStatus::Successful,
-                                       absl::nullopt, absl::nullopt}
-                   : ValidationResults{ValidationResults::ValidationStatus::Failed, tls_alert,
+                                       Envoy::Ssl::ClientValidationStatus::Validated, absl::nullopt,
+                                       absl::nullopt}
+                   : ValidationResults{ValidationResults::ValidationStatus::Failed,
+                                       Envoy::Ssl::ClientValidationStatus::Validated, tls_alert,
```

Comment:
Why is this one `Envoy::Ssl::ClientValidationStatus::Validated`? Should it be `detailed_status`?